tools/ansible-pr/run

```diff
@@ -0,0 +1,230 @@
+  'lib/ansible/module_utils/gcp_utils.py',
+  'lib/ansible/utils/module_docs_fragments/gcp.py'
+].freeze
+
+@options = {
+  directory: "#{Dir.pwd}/build/ansible/",
+  dry_run: false,
+  # Get all PRs from one day ago.
+  date: (Time.now - (3600 * 24)).iso8601
+}
+###########################################################
+# Functions
+
+def git_run(cmd, silent = false, opts = @options)
+  silent_cmd = silent ? '>/dev/null' : ''
+  command = [
+    "pushd #{opts[:directory]} >/dev/null",
+    "#{cmd}#{silent_cmd} 2>&1",
+    "popd #{opts[:directory]} > /dev/null"
+  ]
+
+  if @options[:dry_run]
+    puts command.join('; ')
+  else
+    %x(#{command.join('; ')})
+  end
+end
+
+def new_branch(branch_name, silent = true)
+  git_run('git checkout upstream/devel', silent)
+  git_run("git branch -D #{branch_name}", silent)
+  git_run("git checkout -b #{branch_name}", silent)
+end
+
+# Check that remotes are in order
+def check_remotes(opts = @options)
+  # Check git remotes
+  remotes = git_run('git remote -v', false, opts)
+  remotes = remotes.split("\n")
+                   .map { |line| line.split("\t") }
+                   .map { |line| { line[0] => line[1].split(' ').first } }
+                   .reduce({}, :merge)
+  raise 'Origin git remote should be set' unless remotes['origin']
+  raise "Upstream git remote should be #{ANSIBLE_URL}" unless remotes['upstream'] == ANSIBLE_URL
+end
+
+def get_all_modules(remote)
+  git_run("git fetch #{remote} devel", true)
+  git_run("git checkout #{remote}/devel", true)
+  files = git_run('git ls-files -- lib/ansible/modules/cloud/google/gcp_*')
+  files.split("\n")
+       .map { |line| %r{lib/ansible/modules/cloud/google/([a-z_]*)\.py}.match(line.strip)[1] }
+       .reject { |line| IGNORED_MODULES.include?(line) }
+end
+
+def check_if_different(file, remote1, remote2)
+  @files ||= git_run("git diff --name-only #{remote1}/devel #{remote2}/devel")
+  @files.split("\n")
+        .include?(file)
+end
+
+def checkout_file(file)
+  git_run("git checkout magician/devel -- #{file}", true)
+end
+
+def all_pr_messages
+  # Limit of 50 bug fixes will go into a PR (should never hit this limit)
+  client = Octokit::Client.new(access_token: @options[:creds], per_page: 50)
+  client.list_issues('googlecloudplatform/magic-modules',
+                     labels: 'ansible,downstream-generated',
+                     state: 'closed', since: @options[:date].strip)
+        .map { |pr| { title: pr[:title], url: pr[:html_url] } }
+end
+#############################################################
+# Main
+#############################################################
+
+OptionParser.new do |opts|
+  opts.banner = 'Usage: run [options]'
+
+  opts.on('--dir', '--directory [DIRECTORY]', 'Ansible repo location') do |v|
+    @options[:directory] = v
+  end
+
+  opts.on('-d', '--dry-run', 'Dry run') do |_v|
+    @options[:dry_run] = true
+  end
+
+  opts.on('--date [DATE (as ISO8601 String)]', 'Date for Github sorting') do |v|
```

Comment:
Could you elaborate on this in the help message?  I don't really understand what it means.

Reply:
Done.